IETF Last Call Comments

kc2rxo opened the pull request:
Changes made during IETF Last Call.

kc2rxo committed:
fix: init v26 / start ietf-lc

ad-comment 32 + spelling errors

kc2rxo committed:
fix: spelling and grammer pass

kc2rxo committed:
fix(opsdir): cidr prefix missing 0s

kc2rxo committed:
fix(opsdir): remove abbreviations in title

kc2rxo committed:
init: v27

kc2rxo committed:
fix(opsdir): add example to 6.1.2

kc2rxo committed:
fix(opsdir): add text about abbreviations

boucadair commented on registries.mkd:
```diff
@@ -435,13 +450,13 @@ The interfaces and protocol specifications for registry-registrar interactions a
 Decisions about DNS or registry best practices and other operational matters that influence security SHOULD be made by the CAA, ideally in consultation with local stakeholders.
 
-The guidance above is intended to reduce the likelihood of interoperability problems and/or minimise security and stability concerns. For instance, validation and authentication of DNS responses depends on DNSSEC. If this is not used, entities using DRIP will be vulnerable to DNS spoofing attacks and could be exposed to bogus data. Some regulatory and legal considerations are expected to be simpl ...
+The guidance above is intended to reduce the likelihood of interoperability problems and/or minimize security and stability concerns. For instance, validation and authentication of DNS responses depends on DNSSEC. If this is not used, entities using DRIP will be vulnerable to DNS spoofing attacks and could be exposed to bogus data. Some regulatory and legal considerations are expected to be simpl ...
```
```suggestion
The guidance above is intended to reduce the likelihood of interoperability problems and minimize security and stability concerns. For instance, validation and authentication of DNS responses depends on DNSSEC. If this is not used, entities using DRIP will be vulnerable to DNS spoofing attacks and could be exposed to bogus data. Some regulatory and legal considerations are expected to be simplified by providing a lookup service for access to public information about registered domain names for DETs.
```

boucadair commented on registries.mkd:
```diff
@@ -156,7 +156,7 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
 ## Additional Definitions
 
-This document makes use of the terms (USS, etc.) defined in {{RFC9153}}. Other terms (DIME, Endorsement, etc.) are from {{RFC9434}}, while others (RAA, HDA, etc.) are from {{RFC9374}}.
+This document makes use of the terms/abbreviations (USS, etc.) defined in {{RFC9153}}. Other terms/abbreviations (DIME, Endorsement, etc.) are from {{RFC9434}}, while others (RAA, HDA, etc.) are from {{RFC9374}}.
```
```suggestion
This document makes use of the terms and abbreviations (e.g., USS) defined in {{Section 2.2 of RFC9153}}. Other terms and abbreviations (e.g., DRIP Identity Management Entity (DIME) and Endorsement) are from {{Section 2 of RFC9434}}, while others (e.g., Registered Assigning Authority (RAA) and HHIT Domain Authority (HDA)) are from {{Section 2 of RFC9374}}.
```

boucadair commented on registries.mkd:
```diff
@@ -240,21 +240,21 @@ Registrants                            |
+Authoritative Name Servers use domain names as identifiers and data is stored in Resource Records (RR) with associated RRTypes. This document defines two new RRTypes, one for HHIT metadata (HHIT, {{hhit-rr}}) and another for UAS Broadcast RID information (BRID, {{bcast-rr}}). The former RRType is particularly important as it contains a URI (as part of the certificate) that points to Private Infor ...
 
-DETs, being IPv6 addresses, are to be under `ip6.arpa` (nibble reversed per convention) and MUST to an HHIT RRType. Depending on local circumstances or additional use cases other RRTypes MAY be present. For UAS RID the BRID RRType MUST be present to provide the Broadcast Endorsements defined in {{RFC9575}}.
+DETs, being IPv6 addresses, are to be under `ip6.arpa.` (nibble reversed per convention) and MUST resolve to an HHIT RRType. Depending on local circumstances or additional use cases other RRTypes MAY be present. For UAS RID the BRID RRType MUST be present to provide the Broadcast Endorsements defined in {{RFC9575}}.
```
```suggestion
DETs, being IPv6 addresses, are to be under `ip6.arpa.` (nibble reversed per convention) and MUST resolve to an HHIT RRType. Depending on local circumstances or additional use cases other RRTypes MAY be present. For UAS RID the BRID RRType MUST be present to provide the Broadcast Endorsements (BEs) defined in {{Section 3.1.2.1 of RFC9575}}.
```

boucadair commented on registries.mkd:
```diff
@@ -268,13 +268,13 @@ The HHIT RRType provides the public key for signature verification and URIs via
 ## HHIT Resource Record {#hhit-rr}
 
-The HHIT Resource Record is a metadata record for various bits of HHIT specific information that isn't available in the pre-existing HIP RRType. It does not replace the HIP RRType. The primary advantage of this RRType over the existing RRType is the inclusion a certificate containing an entity's public key signed by the registrar, or other trust anchor, to confirm registration.
+The HHIT Resource Record is a metadata record for various bits of HHIT specific information that isn't available in the pre-existing HIP RRType. It does not replace the HIP RRType. The primary advantage of this RRType over the existing RRType is the inclusion of a certificate containing an entity's public key signed by the registrar, or other trust anchor, to confirm registration.
```
```suggestion
The HHIT Resource Record (RR) is a metadata record for various bits of HHIT specific information that isn't available in the pre-existing HIP RRType. The HHIT RRType does not replace the HIP RRType. The primary advantage of the HHIT RRType over the existing RRType is the inclusion of a certificate containing an entity's public key signed by the registrar, or other trust anchor, to confirm registration.
```

boucadair commented on registries.mkd:
```diff
@@ -289,15 +289,15 @@ The data MAY, for display purposes only, be represented using the Extended Diagn
 HID Abbreviation:
 
-: This field is a string meant to provide an abbreviation to the HID structure for display devices. The convention for such abbreviations is a matter of local policy. Absent of such a policy, this field MUST be filled with the four chracter hexadecimal representations of the RAA and HDA (in that order) with a seperator character such as a space. For example a DET with an RAA value of 10 and HDA v ...
+: The `HID Abbreviation` field is a string meant to provide an abbreviation to the HID structure of a DET for display devices. The convention for such abbreviations is a matter of local policy. Absent of such a policy, this field MUST be filled with the four character hexadecimal representations of the RAA and HDA (in that order) with a separator character such as a space. For example a DET with  ...
```
```suggestion
: The `HID Abbreviation` field is a string that provides an abbreviation to the HID structure of a DET for display devices. The convention for such abbreviations is a matter of local policy. Absent of such a policy, this field MUST be filled with the four character hexadecimal representations of the RAA and HDA (in that order) with a separator character such as a space. For example, a DET with an RAA value of 10 and HDA value of 20 would be abbreviated as: `000A 0014`.
```

kc2rxo committed:
fix: med changes 1

Co-authored-by: Med <mohamed.boucadair@orange.com>

kc2rxo committed:
fix: med changes 2

Co-authored-by: Med <mohamed.boucadair@orange.com>

kc2rxo committed:
fix: med changes 3

Co-authored-by: Med <mohamed.boucadair@orange.com>

kc2rxo committed:
fix: switch to arrays from groups

kc2rxo committed:
feat: rrtype request forms

kc2rxo committed:
fix: template submission

kc2rxo committed:
fix(secdir): require certificate in hhit-rr

HHIT RRType CDDL states that the certificate field is required. Added to text in prose to reinforce this. Also added reference to RFC5280 for DET X.509 encoding.

kc2rxo committed:
fix(secdir): dnssec clarifications

Adds text clarifying when DNSSEC is not used what is expected of a client.

kc2rxo committed:
fix(secdir): det and pk exposure

Reword text to clarify the concept of "just in time" RRType publication.

kc2rxo committed:
fix: med changes 4

Co-authored-by: Med <mohamed.boucadair@orange.com>

kc2rxo committed:
fix: med changes 5

Co-authored-by: Med <mohamed.boucadair@orange.com>

kc2rxo committed:
init: v28

kc2rxo committed:
fix(secdir): added text for dnssec from jr

kc2rxo committed:
fix(secdir): additional text on hash length for dnssec

kc2rxo committed:
fix(rgm): example clarity

boucadair commented on registries.mkd:
```diff
@@ -559,7 +559,7 @@ Certificate:
         64:df:df:60:ef:6f:c0:33:13:0f
 ~~~~
-{: title="7.b.0.a.1.9.e.1.7.5.1.a.0.6.e.5. Decoded HHIT RRType CBOR and Certificate"}
+{: title="2001:3F:FE00:5:5E60:A157:1E91:A0B7 Decoded HHIT RRType CBOR and Certificate"}
```
```suggestion
{: title="2001:3f:fe00:5:5e60:a157:1e91:a0b7 Decoded HHIT RRType CBOR and Certificate"}
```

evyncke replied:
Thank you @boucadair , I should have spotted it ;-)

boucadair commented on registries.mkd:
```diff
@@ -668,11 +668,11 @@ Certificate:
         1e:de:1a:4c:82:8a:ef:84:39:09
 ~~~~
-{: title="0.a.9.0.7.2.4.d.5.4.e.e.5.1.6.6.5.0. Decoded Auth HHIT RRType CBOR and Certificate"}
+{: title="2001:3F:FE00:A05:6615:EE45:D427:9A0 Decoded Auth HHIT RRType CBOR and Certificate"}
```
```suggestion
{: title="2001:3f:fe00:a05:6615:ee45:d427:9a0 Decoded Auth HHIT RRType CBOR and Certificate"}
```

boucadair commented on registries.mkd:
```diff
@@ -726,7 +726,7 @@ Certificate:
         ac:7f:28:b7:45:15:38:e0:80:04
 ~~~~
-{: title="8.2.e.6.5.2.b.6.7.3.4.d.e.0.6.2.5.0. Decoded Issue HHIT RRType CBOR and Certificate"}
+{: title="2001:3F:FE00:A05:260E:D437:6B25:6E28 Decoded Issue HHIT RRType CBOR and Certificate"}
```
```suggestion
{: title="2001:3f:fe00:a05:260e:d437:6b25:6e28 Decoded Issue HHIT RRType CBOR and Certificate"}
```

boucadair commented on registries.mkd:
```diff
@@ -828,7 +828,7 @@ Certificate:
         a4:47:03:79:88:5e:b2:5d:a3:05
 ~~~~
-{: title="2.b.6.c.b.4.a.9.9.6.4.2.8.0.3.1. Decoded HHIT RRType CBOR and Certificate"}
+{: title="2001:3F:FE00:A05:1308:2469:9A4B:C6B2 Decoded HHIT RRType CBOR and Certificate"}
```
```suggestion
{: title="2001:3f:fe00:a05:1308:2469:9a4b:c6b2 Decoded HHIT RRType CBOR and Certificate"}
```

boucadair commented on registries.mkd:
```diff
@@ -878,4 +878,4 @@ Certificate:
 ~~~~
-{: title="2.b.6.c.b.4.a.9.9.6.4.2.8.0.3.1. Decoded BRID RRType CBOR"}
\ No newline at end of file
+{: title="2001:3F:FE00:A05:1308:2469:9A4B:C6B2 Decoded BRID RRType CBOR"}
```
```suggestion
{: title="2001:3f:fe00:a05:1308:2469:9a4b:c6b2 Decoded BRID RRType CBOR"}
```

boucadair reviewed (changes requested):
Please follow rfc5952. Thanks.

kc2rxo committed:
fix(med): rfc5952

Co-authored-by: Med <mohamed.boucadair@orange.com>

kc2rxo committed:
fix(med): rfc5952

Co-authored-by: Med <mohamed.boucadair@orange.com>

kc2rxo committed:
fix(med): rfc5952

Co-authored-by: Med <mohamed.boucadair@orange.com>

kc2rxo committed:
fix(med): rfc5952

Co-authored-by: Med <mohamed.boucadair@orange.com>

kc2rxo committed:
fix(med): rfc5952

Co-authored-by: Med <mohamed.boucadair@orange.com>

kc2rxo committed:
fix(iana): text of templates

kc2rxo committed:
feat(iana): allocated rrtypes

kc2rxo committed:
init: v29

kc2rxo committed:
iesg(gunter): abstract change

kc2rxo committed:
iana: change link to drip registry page

kc2rxo committed:
iana: github url fixed

Now to specific commit not file and inside a note that is marked to be removed by RFC Editor before publication.

kc2rxo committed:
iana: add hex prefix to hhit-type table

kc2rxo committed:
iana: fix wording of ISO/RAA mapping statement

kc2rxo committed:
fix: tweaked abstract to add DIME and DNS terms

kc2rxo committed:
iesg: additional definitions refactor

kc2rxo committed:
iesg(newton): add icao reference to defs

kc2rxo committed:
iesg(newton, comment): non-normative for physical characteristics

kc2rxo committed:
iesg(bishop, comment): nibble reverse reference

kc2rxo committed:
iesg(bishop, comment): canonical reg cert

kc2rxo committed:
iesg(bishop, comment): brid grammer fix

kc2rxo committed:
iesg(bishop, comment): nits

kc2rxo committed:
iesg(wouter, discuss): poor sentence for example

kc2rxo committed:
refactor: break examples into seperate blocks for decoded cbor and cert

kc2rxo committed:
iesg(steel, comment): wrong section of 9374

kc2rxo committed:
fix: section 2.5 std 88 reference

kc2rxo committed:
build: v30

kc2rxo committed:
iesg(bishop-comment): remove sentence

kc2rxo committed:
iesg(newton, discuss): remove whois

kc2rxo committed:
iesg(newton, discuss): dnssec must/recommend

kc2rxo committed:
iesg(newton, comment/wouter, discuss): epp or equivilant

kc2rxo committed:
iesg(wouters, discuss): use bcp237

kc2rxo committed:
iesg(wouters, discuss): example of DS RRType

kc2rxo committed:
iesg(wouters, discuss): better figure explainantions

kc2rxo committed:
iesg(roman, discuss): fix raa table for policy

kc2rxo committed:
iesg(roman, comment): clarify be validation should

kc2rxo committed:
v31

boucadair commented on registries.mkd:
```diff
@@ -261,9 +252,9 @@ Per {{RFC9434}} all information classified as public is stored in the DNS, speci
 Authoritative Name Servers use domain names as identifiers and data is stored in Resource Records (RR) with associated RRTypes. This document defines two new RRTypes, one for HHIT metadata (HHIT, {{hhit-rr}}) and another for UAS Broadcast RID information (BRID, {{bcast-rr}}). The former RRType is particularly important as it contains a URI (as part of the certificate) that points to Private Infor ...
 
-DETs, being IPv6 addresses, are to be under `ip6.arpa.` (nibble reversed per Section 2.5 of {{STD88}}) and MUST resolve to an HHIT RRType. Depending on local circumstances or additional use cases other RRTypes MAY be present. For UAS RID the BRID RRType MUST be present to provide the Broadcast Endorsements (BEs) defined in {{Section 3.1.2.1 of RFC9575}}.
+DETs, being IPv6 addresses, are to be under `ip6.arpa.` (nibble reversed per Section 2.5 of {{STD88}}) and MUST resolve to an HHIT RRType. Depending on local circumstances or additional use cases other RRTypes MAY be present (for example the inclusion of the DS RRTypes or equivalent when using DNSSEC). For UAS RID the BRID RRType MUST be present to provide the Broadcast Endorsements (BEs) defined ...
```
```suggestion
DETs, being IPv6 addresses, are to be under `ip6.arpa.` (nibble reversed per Section 2.5 of {{STD88}}) and MUST resolve to an HHIT RRType. Depending on local circumstances or additional use cases other RRTypes MAY be present (for example, the inclusion of the DS RRTypes or equivalent when using DNSSEC). For UAS RID the BRID RRType MUST be present to provide the Broadcast Endorsements (BEs) defined in {{Section 3.1.2.1 of RFC9575}}.
```

boucadair commented on registries.mkd:
```diff
@@ -355,14 +346,14 @@ This document requests a new registry for RAA Allocations under the [DRIP regist
+| RAA Range     | Allocation                 | Policy                                             |
+| ------------- | -------------------------- | -------------------------------------------------- |
+| 0 - 3         | Reserved                   | N/A                                                |
+| 4 - 3999      | ISO 3166-1 Countries       | Expert Review {{Section 4.5 of RFC8126}}           |
```
```suggestion
| 4 - 3999      | ISO 3166-1 Countries       | Expert Review ({{Section 4.5 of RFC8126}})           |
```

boucadair commented on registries.mkd:
```diff
@@ -355,14 +346,14 @@ This document requests a new registry for RAA Allocations under the [DRIP regist
+| 0 - 3         | Reserved                   | N/A                                                |
+| 4 - 3999      | ISO 3166-1 Countries       | Expert Review {{Section 4.5 of RFC8126}}           |
+| 4000 - 15359  | Reserved                   | N/A                                                |
+| 15360 - 16383 | Experimental Use           | First Come First Served {{Section 4.4 of RFC8126}} |
```
```suggestion
| 15360 - 16383 | Experimental Use           | First Come First Served ({{Section 4.4 of RFC8126}}) |
```

boucadair commented on registries.mkd:
```diff
@@ -407,8 +398,8 @@ HHIT Entity Type:
 | ----------------------- | ------------------------------------------- | ---------------------- |
-| 0x00000000              | 0 - 4294967295                              | Expert Review (Section 4.5 of {{RFC8126}}) |
-| 0x00000001 - 0xFFFFFFFE | 4294967296 - 18446744069414584319           | First Come First Served (Section 4.4 of {{RFC8126}}) |
+| 0x00000000              | 0 - 4294967295                              | Expert Review {{Section 4.5 of RFC8126}} |
```
```suggestion
| 0x00000000              | 0 - 4294967295                              | Expert Review ({{Section 4.5 of RFC8126}}) |
```

boucadair commented on registries.mkd:
```diff
@@ -407,8 +398,8 @@ HHIT Entity Type:
-| 0x00000000              | 0 - 4294967295                              | Expert Review (Section 4.5 of {{RFC8126}}) |
-| 0x00000001 - 0xFFFFFFFE | 4294967296 - 18446744069414584319           | First Come First Served (Section 4.4 of {{RFC8126}}) |
+| 0x00000000              | 0 - 4294967295                              | Expert Review {{Section 4.5 of RFC8126}} |
+| 0x00000001 - 0xFFFFFFFE | 4294967296 - 18446744069414584319           | First Come First Served {{Section 4.4 of RFC8126}} |
```
```suggestion
| 0x00000001 - 0xFFFFFFFE | 4294967296 - 18446744069414584319           | First Come First Served ({{Section 4.4 of RFC8126}}) |
```

boucadair commented on registries.mkd:
```diff
@@ -435,6 +426,7 @@ The following values are defined by this document:
 | 24            | Network RID Service Provider (SP)          | This RFC     |
 | 25            | Network RID Display Provider (DP)          | This RFC     |
 | 26            | Supplemental Data Service Provider (SDSP)  | This RFC     |
+| 27            | Crowd Sourced RID Finder                   | This RFC     |
```
I'm afraid this entry is not justified here as the intended use is underspecified.

kc2rxo committed:
iesg(eric): DET prefix change

kc2rxo committed:
eisg(eric): rework raa iana section

kc2rxo committed:
iesg(eric): rework hhit iana section

kc2rxo committed:
fix: remove ambiguity of exhaustive list

kc2rxo committed:
fix: update iana raa

kc2rxo committed:
iesg: last fixes
